lib/Sema/TypeCheckType.cpp

```diff
@@ -382,48 +382,50 @@ Type TypeChecker::getStringType(ASTContext &Context) {
   if (auto typeDecl = Context.getStringDecl())
     return typeDecl->getDeclaredInterfaceType();
 
-  return Type();
+  llvm_unreachable("Broken Standard library: TypeChecker::getStringType failed.");
 }
 
 Type TypeChecker::getSubstringType(ASTContext &Context) {
   if (auto typeDecl = Context.getSubstringDecl())
     return typeDecl->getDeclaredInterfaceType();
 
-  return Type();
+  llvm_unreachable("Broken Standard library: TypeChecker::getSubstringType failed.");
 }
 
 Type TypeChecker::getIntType(ASTContext &Context) {
   if (auto typeDecl = Context.getIntDecl())
     return typeDecl->getDeclaredInterfaceType();
 
-  return Type();
+  llvm_unreachable("Broken Standard library: TypeChecker::getIntType failed.");
 }
 
 Type TypeChecker::getInt8Type(ASTContext &Context) {
   if (auto typeDecl = Context.getInt8Decl())
     return typeDecl->getDeclaredInterfaceType();
 
-  return Type();
+  llvm_unreachable("Broken Standard library: TypeChecker::getInt8Type failed.");
 }
 
 Type TypeChecker::getUInt8Type(ASTContext &Context) {
   if (auto typeDecl = Context.getUInt8Decl())
     return typeDecl->getDeclaredInterfaceType();
 
-  return Type();
+  llvm_unreachable("Broken Standard library: TypeChecker::getUInt8Type failed.");
 }
 
 Type
 TypeChecker::getDynamicBridgedThroughObjCClass(DeclContext *dc,
                                                Type dynamicType,
                                                Type valueType) {
   // We can only bridge from class or Objective-C existential types.
+  // if we return ErrorType here, callers will continue trying to type check
+  // and return diagnostic if it fails.
```

Comment:
This is not an invariant we have control over from this interface. If we can't diagnose here, we shouldn't make this transformation unless we update the callers to be more defensive.

Reply:
The original logic had TypeChecker::getDynamicBridgedThroughObjCClass returning Type() if the bridge failed, and the client (repairFailures) continuing to try to match the type. Returning ErrorType is a little clunky because it's really indicating a type match failure in this leaf of the tree. It does not return a diagnostic (violating the contract of ErrorType) because it's not a diagnostic-worthy error yet. If the whole matchType->repairFailures process fails then you get a diagnostic. That was what I meant in the comment but it's not that clear. If there were a 'NoMatchType' that we could return for this situation it might be clearer but that's also a wider change and adds state to the system.